### DISCUSSION TITLE
Question about Heartbeat [13643]

### DISCUSSION THREAD

**Discussion by advancenXX** [eProsima/Fast-DDS#2445] [Q&A]
I have about 100 dds nodes within the same domain in my device which causes too many Heartbeat packages.
How could I decrease it? Increasing [Wparam.times.heartbeatPeriod.seconds] seems have little impact.

**Comment by MiguelCompany** [comment]
The parameter you are setting refers only to the heartbeats produced by the writer to which you are passing that QoS setting.

If you want to reduce the number of heartbeats of the built-in writers, you could try setting [avoid_builtin_multicast](https://fast-dds.docs.eprosima.com/en/latest/fastdds/xml_configuration/domainparticipant.html?highlight=avoid_builtin_multicast#builtin-parameters) to `false` on the `DomainParticipant` configuration

**Comment by MiguelCompany** [comment]
@advancenXX Also note that `heartbeatPeriod` does not force a heartbeat to be sent. It only sets the period for the writer to check if a heartbeat needs to be sent (depending on the state of the readers).

**Comment by advancenXX** [comment]
> If you want to reduce the number of heartbeats of the built-in writers, you could try setting [avoid_builtin_multicast](https://fast-dds.docs.eprosima.com/en/latest/fastdds/xml_configuration/domainparticipant.html?highlight=avoid_builtin_multicast#builtin-parameters) to `false` on the `DomainParticipant` configuration

OK, I will try it. But why avoid_builtin_multicast is related to Heartbeat? It seems like something about PDP, a StatelessWriter.

**Comment by MiguelCompany** [comment]
Setting it to `false` will enable multicast in the rest of the builtin endpoints (i.e. EDP, WLP), which are StatefulWriter ones. Using multicast on them will allow sending the same heartbeat once, instead of repeating the same heartbeat to the builtin reader of every participant.

**Comment by advancenXX** [comment]
OK. And there are my parameters, 
    avoid_builtin_multicast = false;
    heartbeatPeriod = 30 sec;
    disable_heartbeat_piggyback = true;
Do you have any other suggestion? As this is a real big problem for my device.

**Comment by advancenXX** [comment]
Hi @MiguelCompany , after setting avoid_builtin_multicast to false, there are still many unicast packets even though multicast is enabled.
How could I disable all unicast in builtin EDP and WLP?
Thanks a lot.
